DeeperFunc/String.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  String.swift
+//  DeeperFunc
+//
+//  Created by Ilya Puchka on 28/10/2017.
+//  Copyright © 2017 Ilya Puchka. All rights reserved.
+//
+
+import Foundation
+
+func and(_ lhs: RoutePattern<Void, Path>, _ rhs: RoutePattern<Void, Path>) -> RoutePattern<Void, Path> {
+    return .init(parse: parseRight(lhs, rhs), print: printRight(lhs, rhs), template: templateAnd(lhs, rhs))
+}
+
+extension String {
```

Comment:
all these operators could be probably replaced with conditional conformance, then maybe `/` could be used without creating too complex expressions